dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -323,18 +344,17 @@ void DeltaMergeStore::setUpBackgroundTask(const DMContextPtr & dm_context)
                     continue;
 
                 // Note that page_id is useless here.
-                auto dmfile = DMFile::restore(global_context.getFileProvider(), id, /* page_id= */ 0, path, DMFile::ReadMetaMode::none());
+                auto dmfile = DMFile::restore(file_provider, id, /* page_id= */ 0, path, DMFile::ReadMetaMode::none());
                 if (dmfile->canGC())
                 {
                     delegate.removeDTFile(dmfile->fileId());
```

Comment:
If this method is called during the `DeltaMergeStore::drop()`, I think it will rise `Cannot find DMFile` exception.

Reply:
I'll try a better way to handle it